linfa-clustering/src/k_means/hyperparameters.rs

```diff
@@ -30,13 +32,23 @@ pub struct KMeansHyperParams<F: Float, R: Rng> {
 /// An helper struct used to construct a set of [valid hyperparameters](struct.KMeansHyperParams.html) for
 /// the [K-means algorithm](struct.KMeans.html) (using the builder pattern).
 pub struct KMeansHyperParamsBuilder<F: Float, R: Rng> {
+    n_init: u64,
     tolerance: F,
     max_n_iterations: u64,
     n_clusters: usize,
     rng: R,
 }
 
 impl<F: Float, R: Rng + Clone> KMeansHyperParamsBuilder<F, R> {
+    /// Set the value of `n_init`.
+    ///
+    /// The final results will be the best output of n_init consecutive runs in terms of inertia
+    /// (sum of squared distances to the closest centroid for all observations in the training set)
+    pub fn n_init(mut self, n_init: u64) -> Self {
```

Comment:
I think that `n_init` suggests that some pre-initialization happens before the actual K-Means. What do you think about a name like `num_runs`?

Reply:
I should have referenced the [scikit-learn implementation](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html). So I think it is better to keep the same name (it si also used in GMM).

Reply:
... But I understand your point. In that case I would go for <code>n_runs</code> for consistency and change it also in GMM. What do you think?

Reply:
sounds good to me :+1: we don't have to follow Scikit-Learn here and probably should try to improve the interface